gpt_slide_explainer.py

```diff
@@ -0,0 +1,115 @@
+import sys
+from typing import Dict
+import asyncio
+import openai
+
+MAX_TOKENS = 200
+TEMPERATURE = 0.7
+
+WINDOWS_PLATFORM = 'win'
+
+
+class GPTSlideExpander:
+
+    def __init__(self):
+        GPTSlideExpander.init_api_key()
+        self.expanded_slide_explanations = {}
+        if sys.platform.startswith(WINDOWS_PLATFORM):
+            asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+
+    @staticmethod
+    def init_api_key():
+        """get the api key from file"""
+        with open(r"C:\Users\USER\Downloads\api_key.txt", 'r') as f:
+            openai.api_key = str(f.read())
+
+    async def generate_explanation_for_presentation(self, parsed_presentation: Dict[int, str], presentation_topic: str
+                                                    ) -> None:
+        """
+        Generates explanations for each slide in a presentation.
+        Args:
+            parsed_presentation (Dict[int, str]): A dictionary mapping slide index to slide content.
+            presentation_topic (str): main topic of presentation.
+        Raises:
+            RuntimeError: If the main topic retrieval from the API fails.
+        Returns:
+            None
+        """
+
+        async def process_slide(index: int, slide):
```

Comment:
You choice to use an inner an inner function assumes the you wont need it anywhere else in the future. I wouldn't neccerilly do the same choice, in this situations. Anyways, you can read this -

Reply:
https://realpython.com/inner-functions-what-are-they-good-for/